### PR TITLE
WIP:[NALM/449] Refactor event system

### PR DESCRIPTION
**Pull request by jogboms**

### CODE REVIEW

File: src/www/scripts/app.js [hunk 1]
```diff
@@ -10,6 +9,23 @@ import SecureStore from "./secure-store";
...
 
+function Emitter() {
+    let events = {};
+    return {
+        on: (eventName, callback) => {
+            events[eventName] = [...(events[eventName] || []), callback];
+            return this;
+        },
+        emit: async (eventName, ...args) => {
+            const callbacks = events[eventName];
```
Comment (KevinVlaanderen) [inline src/www/scripts/app.js]: 1. If you add a fallback to [] here (as in `|| []`)

File: src/www/scripts/app.js [hunk 2]
```diff
@@ -10,6 +9,23 @@ import SecureStore from "./secure-store";
...
+            if (callbacks && callbacks.length > 0) {
+                return await Promise.all(callbacks.map(callback => callback(...args)));
```
Comment (KevinVlaanderen) [inline src/www/scripts/app.js]: 2. Then you can just `return Promise.all` here

File: src/www/scripts/app.js [hunk 3]
```diff
@@ -10,6 +9,23 @@ import SecureStore from "./secure-store";
...
+function Emitter() {
+    let events = {};
+    return {
+        on: (eventName, callback) => {
+            events[eventName] = [...(events[eventName] || []), callback];
+            return this;
+        },
+        emit: async (eventName, ...args) => {
+            const callbacks = events[eventName];
+            if (callbacks && callbacks.length > 0) {
```
Comment (KevinVlaanderen) [inline src/www/scripts/app.js]: 3.a And get rid of the whole `if` block

File: src/www/scripts/app.js [hunk 4]
```diff
@@ -10,6 +9,23 @@ import SecureStore from "./secure-store";
...
+                return await Promise.all(callbacks.map(callback => callback(...args)));
+            }
+            return Promise.resolve();
```
Comment (KevinVlaanderen) [inline src/www/scripts/app.js]: 3.b ...and the `Promise.resolve`

**Review by KevinVlaanderen — CHANGES_REQUESTED** [review]

File: src/www/scripts/app.js (hunk 1 above)
Comment (jogboms) [reply src/www/scripts/app.js]: Done

File: src/www/scripts/app.js (hunk 2 above)
Comment (jogboms) [reply src/www/scripts/app.js]: Done

File: src/www/scripts/app.js (hunk 3 above)
Comment (jogboms) [reply src/www/scripts/app.js]: Done

File: src/www/scripts/app.js (hunk 4 above)
Comment (jogboms) [reply src/www/scripts/app.js]: Done

**Review by KevinVlaanderen — APPROVED** [review]

**Pull request merged**